nerdtree_plugin/file_filter.vim

```diff
@@ -28,36 +28,78 @@ if !exists('g:NERDTreeFileFilterEnabled')
     let g:NERDTreeFileFilterEnabled = 0
 endif
 
+function! NERDTreeDirectoryFilter(params)
+    if g:NERDTreeFileFilterEnabled != 1
+        return 0
+    endif
+
+    let path = a:params['path']
+
+    if path.isDirectory
+        let tree = a:params['nerdtree']
+        let node = b:NERDTreeRoot.findNode(path)
+        if empty(node)
+            return 0
+        endif
+
+        return s:noVisibleChildren(node, tree)
+    endif
+
+    return 0
+endfunction
+
+function! s:noVisibleChildren(node, tree)
+  return !s:anyVisibleChildren(a:node, a:tree)
+endfunction
+
+function! s:anyVisibleChildren(node, tree)
+    for i in a:node.children
+        if !i.path.ignore(a:tree)
```

Comment:
I need a way of finding files that have been filtered, but this just shows files that are not currently visible.
Which is the same thing when the tree is fully expanded, but if you have unexpanded nodes when you filter, then you won't see matching files in those unexpanded nodes.
Which isn't really desirable.